### PR TITLE
Feat/wagmi

### PR DESCRIPTION
**Pull request by msadura**
### <samp>🤖 Generated by Copilot at 62c706e</samp>

The pull request refactors the web3 connection and interaction logic across the app by using the `viem` and `wagmi` libraries, which provide simplified and standardized interfaces for web3 providers, signers, and errors. The pull request also improves the code organization, readability, and consistency by moving and renaming files and hooks, and removing unused variables and imports. Additionally, the pull request enables the user to connect an identity provider to their web3 account from the new identity modal.

### WHY
Move from web3-react to wagmi

### CODE REVIEW

File: theme/styles.scss [hunk 1]
```diff
@@ -111,4 +111,9 @@ input:-webkit-autofill:active {
   .MuiButton-root {
     font-family: var(--font-family-default);
   }
+}
+
+body {
+  /* wallet connect modal */
+  --wcm-z-index: 1500;
```
Comment (msadura) [inline theme/styles.scss:118]: this was needed to display wallet connect modal above MUI's modal

File: .storybook/preview.tsx [hunk 2]
```diff
@@ -138,13 +140,21 @@ export const withMuiTheme = (Story, context) => {
   );
 };
 
+const { publicClient } = configureChains([mainnet], [publicProvider()]);
```
Comment (motechFR) [inline .storybook/preview.tsx:143]: I thought we'd need more than mainnet, but was able to make a transaction on optimisim

File: components/_app/Web3ConnectionManager/components/WalletSelectorModal/WalletSelectorModal.tsx [hunk 3]
```diff
@@ -84,40 +68,39 @@ export function WalletSelector({ loginSuccess = () => null, onError = () => null
...
           />
         </Grid>
 
         <Grid item xs={12}>
           <ConnectorButton
             name='WalletConnect'
             onClick={() => {
-              WalletConnectV2Connector.clearStorage(window.localStorage);
-              handleConnect(walletConnect);
+              handleConnect(walletConnectConnector);
```
Comment (motechFR) [inline components/_app/Web3ConnectionManager/components/WalletSelectorModal/WalletSelectorModal.tsx:88]: I don't remember why we cleared local storage, I think it was to do with outdated preferences.

File: components/bounties/components/MultiPaymentModal.tsx [hunk 4]
```diff
@@ -40,7 +40,7 @@ export function MultiPaymentModal({ bounties }: { bounties: BountyWithDetails[]
   const { data: userGnosisSafes } = useMultiWalletSigs();
   const { importSafes } = useImportSafes();
   const { onClick } = useSettingsDialog();
-  const gnosisSigner = useGnosisSigner();
+  const { signer } = useWeb3Signer();
```
Comment (motechFR) [inline components/bounties/components/MultiPaymentModal.tsx:43]: Suggestion; Refactor signer into useWeb3AuthSig (we could also rename this to useWeb3)

File: components/common/PageActions/components/SnapshotAction/PublishingForm.tsx [hunk 5]
```diff
@@ -210,7 +212,7 @@ export function PublishingForm({ onSubmit, pageId }: Props) {
         generatorOptions: { members }
       });
 
-      if (!account) {
+      if (!account || !web3Provider) {
```
Comment (motechFR) [inline components/common/PageActions/components/SnapshotAction/PublishingForm.tsx:215]: Probably could just come from useWeb3

File: components/login/components/WalletSign.tsx [hunk 6]
```diff
@@ -31,16 +32,8 @@ export function WalletSign({
...
-    account,
-    sign,
-    isSigning,
-    walletAuthSignature,
-    verifiableWalletDetected,
-    connectWallet,
-    connectWalletModalIsOpen,
-    isConnectingIdentity
-  } = useWeb3AuthSig();
+  const { isConnectingIdentity, connectWallet, isWalletSelectorModalOpen } = useWeb3ConnectionManager();
```
Comment (motechFR) [inline components/login/components/WalletSign.tsx:35]: This is a good one, I like that we finally slimmed down web3AuthSig

File: hooks/useMultiBountyPayment.ts [hunk 7]
```diff
@@ -41,7 +41,7 @@ export function useMultiBountyPayment({
   const { refreshBounties } = useBounties();
   const { account, chainId } = useWeb3AuthSig();
   const [paymentMethods] = usePaymentMethods();
-  const signer = useGnosisSigner();
+  const { signer } = useWeb3Signer();
   const { data: gnosisSafes } = useSWR(
     signer && account && chainId ? `/connected-gnosis-safes/${account}` : null,
```
Comment (motechFR) [inline hooks/useMultiBountyPayment.ts:46]: I managed to get an error when using Gnosis Safe multi payment in bounties

Screenshot 2023-09-14 at 18 58 20

**Review by motechFR — APPROVED** [review]
Looks great! Glad we got this one done.

I noted a bug on Gnosis, let me know if this happens for you as well.

Would also be good to test for Lens Publication

File: components/bounties/components/MultiPaymentModal.tsx (hunk 4 above)
Comment (msadura) [reply components/bounties/components/MultiPaymentModal.tsx:43]: oh IU like putting signer there, nice idea

**Pull request merged**